src/mappings/entities/mapIdentities.ts

```diff
@@ -0,0 +1,250 @@
+import { Codec } from '@polkadot/types/types';
+import { SubstrateEvent } from '@subql/types';
+import { AssetPermissions, PortfolioPermissions, TransactionPermissions } from '../../types';
+import { Account } from '../../types/models/Account';
+import { Identity } from '../../types/models/Identity';
+import { Permissions } from '../../types/models/Permissions';
+import { getFirstValueFromJson, getTextValue } from '../util';
+import { EventIdEnum, ModuleIdEnum } from './common';
+
+/**
+ * Subscribes to the Identities related events
+ */
+export async function mapIdentities(
+  blockId: number,
+  eventId: EventIdEnum,
+  moduleId: ModuleIdEnum,
+  params: Codec[],
+  event: SubstrateEvent
+): Promise<void> {
+  const datetime = event.block.timestamp;
+
+  if (moduleId === ModuleIdEnum.Identity) {
+    if (eventId === EventIdEnum.DidCreated) {
+      await handleDidCreated(blockId, eventId, params, datetime);
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysAdded) {
+      await handleSecondaryKeysAdded(blockId, eventId, params, datetime);
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysFrozen) {
+      await handleSecondaryKeysFrozen(blockId, eventId, params, true);
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysUnfrozen) {
+      await handleSecondaryKeysFrozen(blockId, eventId, params, false);
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysRemoved) {
+      await handleSecondaryKeysRemoved(params);
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeyPermissionsUpdated) {
+      await handleSecondaryKeysPermissionsUpdated(blockId, params);
+    }
+  }
+}
+
+const fetchIdentity = async (param: Codec): Promise<Identity> => {
```

Comment:
(nit) I think similar functions use `get` e.g. `getAsset`.

```suggestion
const getIdentity = async (param: Codec): Promise<Identity> => {
```

I'd be OK with a convention of `fetch` meaning async and `get` being a sync getter like `getTextValue` if thats what you are thinking, but my preference would be to keep everything with `get` since TS will let us know if something is async or not.

Reply:
Makes sense to me.